features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Operations/Operations.jsx

```diff
@@ -0,0 +1,444 @@
+        marginLeft: theme.spacing.unit * 2,
+        textTransform: theme.custom.leftMenuTextStyle,
+        color: theme.palette.getContrastText(theme.palette.primary.main),
+    },
+    buttonMain: {
+        textTransform: theme.custom.leftMenuTextStyle,
+        color: theme.palette.getContrastText(theme.palette.primary.main),
+    },
+    addNewWrapper: {
+        backgroundColor: theme.palette.background.paper,
+        color: theme.palette.getContrastText(theme.palette.background.paper),
+        border: 'solid 1px ' + theme.palette.grey['300'],
+        borderRadius: theme.shape.borderRadius,
+        marginTop: theme.spacing.unit * 2,
+    },
+    contentWrapper: {
+        maxWidth: theme.custom.contentAreaWidth,
+    },
+    addNewHeader: {
+        padding: theme.spacing.unit * 2,
+        backgroundColor: theme.palette.grey['300'],
+        fontSize: theme.typography.h6.fontSize,
+        color: theme.typography.h6.color,
+        fontWeight: theme.typography.h6.fontWeight,
+    },
+    addNewOther: {
+        padding: theme.spacing.unit * 2,
+    },
+    radioGroup: {
+        display: 'flex',
+        flexDirection: 'row',
+        width: 300,
+    },
+    addResource: {
+        width: 600,
+        marginTop: 0,
+    },
+    buttonIcon: {
+        marginRight: 10,
+    },
+    expansionPanel: {
+        marginBottom: theme.spacing.unit,
+    },
+    expansionPanelDetails: {
+        flexDirection: 'column',
+    },
+});
+
+/**
+ * This class defined for operation List
+ */
+class Operations extends React.Component {
+    /**
+     *
+     * @param {*} props the props parameters
+     */
+    constructor(props) {
+        super(props);
+        this.state = {
+            swagger: {},
+            scopes: [],
+            notFound: false,
+            showScopes: false,
+            apiPolicies: [],
+            operationList: null,
+        };
+
+        this.newApi = new Api();
+        this.handleUpdateList = this.handleUpdateList.bind(this);
+        this.toggleAssignScopes = this.toggleAssignScopes.bind(this);
+        this.handleScopeChange = this.handleScopeChange.bind(this);
+        this.updateOperations = this.updateOperations.bind(this);
+        this.handleScopeChangeInSwaggerRoot = this.handleScopeChangeInSwaggerRoot.bind(this);
+    }
+
+    /**
+     *
+     */
+    componentDidMount() {
+        this.setState({
+            operationList: this.props.api.operations,
+            scopes: this.props.api.scopes,
+        });
+        const promisedApi = this.newApi.getSwagger(this.props.api.id);
+        promisedApi
+            .then((response) => {
+                let tempScopes = [];
+                if (response.obj.security && response.obj.security.length !== 0) {
+                    response.obj.security.map((object) => {
+                        if (object.OAuth2Security) {
+                            tempScopes = object.OAuth2Security;
+                        }
+                        return tempScopes;
```

Comment:
i think we can use javascript filter here

`
tempScopes = response.obj.security.filter(object=> object.OAuth2Security)
`

Is there any specific reason we didnt use filter ?